modin/config/envvars.py

```diff
@@ -851,4 +851,11 @@ def _check_vars() -> None:
         )
 
 
+class UsePlainPandasQueryCompiler(EnvironmentVariable, type=bool):
```

Comment:
This name is probably a little confusing for users. I suggest something like `SmallDataframeMode`. This can be set to `None` by default, and users can set it to `"pandas"` or some other option in the future (we may have some other single node options coming).

Reply:
In the future we may add `polars` mode. If this happens, we might also want to have an option for that. Making it a string keeps it open to other options. If we have pandas in the name, we can only use that mode for pandas execution. I'm open to other names, but I think we don't want to keep adding more and more configs if we have more options later.

Reply:
Doesn't this sound like we may have multiple storage formats for a single execution? Do we really want to support this in future?

Reply:
@devin-petersohn, do you think we could support automatic initialization with small qc depending on a data size threshold in future?

Reply:
I propose to rename `UsePlainPandasQueryCompiler` to `NativeDataframeMode` and `SmallQueryCompiler` to `NativeQueryCompiler` by sort of analogy with `HdkOnNative` we had previously.

Reply:
At a minimum, a more complete definition of this class in the docstring is required.

Reply:
I will update the name to UsePlainPandasQueryCompiler to NativeDataframeMode and SmallQueryCompiler to NativeQueryCompiler.